test/base/src/androidTest/kotlin/io/realm/test/shared/MutableRealmTests.kt

```diff
@@ -61,13 +67,256 @@ class MutableRealmTests {
+            assertEquals(255, shortField)
+            assertEquals(255, intField)
+            assertEquals(1024, longField)
+            assertEquals(false, booleanField)
+            assertEquals(42.42f, floatField)
+            assertEquals(42.42, doubleField)
+            assertEquals(RealmInstant.fromEpochSeconds(42, 42), timestampField)
+
+            // FIXME Lacks nullable types
+            assertEquals(primaryKey, child!!.primaryKey)
+
+            assertEquals("UPDATED", stringListField[0])
+            assertEquals(0x10, byteListField[0])
+            assertEquals('b', charListField[0])
+            assertEquals(255, shortListField[0])
+            assertEquals(255, intListField[0])
+            assertEquals(1024, longListField[0])
+            assertEquals(false, booleanListField[0])
+            assertEquals(3.14f, floatListField[0])
+            assertEquals(3.14, doubleListField[0])
+            assertEquals(RealmInstant.fromEpochSeconds(42, 42), timestampListField[0])
+            assertEquals(primaryKey, objectListField[0].primaryKey)
+
+            assertEquals(null, nullableStringListField[0])
+            assertEquals(null, nullableByteListField[0])
+            assertEquals(null, nullableCharListField[0])
+            assertEquals(null, nullableShortListField[0])
+            assertEquals(null, nullableIntListField[0])
+            assertEquals(null, nullableLongListField[0])
+            assertEquals(null, nullableBooleanListField[0])
+            assertEquals(null, nullableFloatListField[0])
+            assertEquals(null, nullableDoubleListField[0])
+            assertEquals(null, nullableTimestampListField[0])
+        }
+    }
+
+    @Test
+    fun copyToRealm_updatePolicy_all_cyclicObject() {
+        val sample1 = SampleWithPrimaryKey().apply {
+            primaryKey = 1
+            stringField = "One"
+        }
+        val sample2 = SampleWithPrimaryKey().apply {
+            primaryKey = 2
+            stringField = "Two"
+        }
+        sample1.child = sample2
+        sample2.child = sample1
+        realm.writeBlocking {
+            copyToRealm(sample1)
+        }.run {
+            assertEquals(1, primaryKey)
+            assertEquals("One", stringField)
+            child?.run {
+                assertEquals(2, primaryKey)
+                assertEquals("Two", stringField)
+            }
+        }
+
+        sample1.stringField = "Three"
+        sample2.stringField = "Four"
+
+        realm.writeBlocking {
+            copyToRealm(sample1, MutableRealm.UpdatePolicy.ALL)
+        }.run {
+            assertEquals(1, primaryKey)
+            assertEquals("Three", stringField)
+            child?.run {
+                assertEquals(2, primaryKey)
+                assertEquals("Four", stringField)
+            }
+        }
+    }
+
+    @Test
+    fun copyToRealm_updatePolicy_all_nonPrimaryKeyObject() {
+        realm.writeBlocking {
+            copyToRealm(Parent(), MutableRealm.UpdatePolicy.ALL)
```

Comment:
Shouldn't we assert that the object is actually inserted?

Reply:
Test is actually overlapping with `copyToRealm_updatePolicy_all_noPrimaryKeyField` that checks insertion.

Reply:
True, maybe just add a comment about that?

Reply:
I just removed this redundant one 😄